libnavui-shield/src/main/java/com/mapbox/navigation/ui/shield/internal/api/MapboxRouteShieldApiEx.kt

```diff
@@ -0,0 +1,19 @@
+package com.mapbox.navigation.ui.shield.internal.api
+
+import com.mapbox.bindgen.Expected
+import com.mapbox.navigation.ui.shield.api.MapboxRouteShieldApi
+import com.mapbox.navigation.ui.shield.internal.model.RouteShieldToDownload
+import com.mapbox.navigation.ui.shield.model.RouteShieldError
+import com.mapbox.navigation.ui.shield.model.RouteShieldResult
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+suspend fun MapboxRouteShieldApi.getRouteShieldsFromModels(
+    shieldsToDownload: List<RouteShieldToDownload>
+): List<Expected<RouteShieldError, RouteShieldResult>> {
+    return suspendCoroutine { continuation ->
```

Comment:
Capturing from @Zayankovsky 

> If inner task cannot be canceled, then its result will just be ignored, but the coroutine will not wait for it.

```suggestion
    return suspendCancellableCoroutine { continuation ->
```

Reply:
I see that first we wrap suspend call with a callback function, then we wrap this callback function with a suspend one. 
Which makes me wonder why don't we expose the suspend function in the first place?

Reply:
> Which makes me wonder why don't we expose the suspend function in the first place?

https://github.com/mapbox/mapbox-navigation-android/issues/4126